lib/src/registry/registry.dart

```diff
@@ -4,33 +4,34 @@
 
 library pointycastle.src.registry;
 
+import "package:pointycastle/api.dart";
+import 'package:pointycastle/src/registry/registration.dart';
 
-import "registry_disabled.dart"
-  if (dart.library.mirrors) "registry_reflectable.dart";
-export  "registry_disabled.dart"
-  if (dart.library.mirrors) "registry_reflectable.dart";
-
-final FactoryRegistry registry = makeRegistry();
+final FactoryRegistry registry = _RegistryImpl();
 
 abstract class FactoryRegistry {
-  dynamic /* Registrable */ create(Type type, String registrableName);
+  T create<T>(String registrableName);
+
+  void register<T>(FactoryConfig config);
 }
 
 typedef /*Registrable*/ RegistrableConstructor();
 typedef RegistrableConstructor DynamicConstructorFactory(
     String registrableName, Match match);
 
-
 abstract class FactoryConfig {
   final Type type;
+
   FactoryConfig(this.type);
 }
 
 class StaticFactoryConfig extends FactoryConfig {
   final String algorithmName;
-  StaticFactoryConfig(Type type, this.algorithmName) : super(type);
-}
+  final RegistrableConstructor factory;
```

Comment:
since we don't use reflection, StaticFactoryConfig needs a function to instantiate actual algorithm instance.